transfer/basic_download.go

```diff
@@ -203,7 +205,11 @@ func (a *basicDownloadAdapter) download(t *Transfer, cb TransferProgressCallback
 	}
 	written, err := tools.CopyWithCallback(dlFile, hasher, res.ContentLength, ccb)
 	if err != nil {
-		return fmt.Errorf("cannot write data to tempfile %q: %v", dlfilename, err)
+		wrappedErr := fmt.Errorf("cannot write data to tempfile %q: %v", dlfilename, err)
+		if errutil.IsRetriableError(err) {
+			return errutil.NewRetriableError(wrappedErr)
+		}
+		return wrappedErr
```

Comment:
👋  thanks for the ping. Handling of errors has changed significantly since v1.3.1, most of the work change are contained in https://github.com/github/git-lfs/pull/1463 and https://github.com/github/git-lfs/pull/1466. The gist of it is that we're now using Dave Cheney's `errors` package, which is described [here](https://godoc.org/github.com/pkg/errors).

How does this one feel?

```go
written, err := tools.CopyWithCallback(...)
if err != nil {
        err = errors.Wrapf(err, "cannot write data to tempfile %q", dlfilename)

        return errors.NewRetriableError(err)
}

// ...
```